From 7c50dba2369e798ca910a75675427084c79c4c0b Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Wed, 6 Aug 2014 03:50:59 -0400
Subject: [pci] Change pci_bus_alloc_resource() type_mask to unsigned long

Message-id: <20140806035059.1438.22668.stgit@gir.stowe>
Patchwork-id: 87473
O-Subject: [RHEL7.1 PATCH 64/69] PCI: Change pci_bus_alloc_resource() type_mask to unsigned long
Bugzilla: 1110896
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110896
Upstream Status: 664c28480c90fb8541bcdd1d4b349e9436165ec7

commit 664c28480c90fb8541bcdd1d4b349e9436165ec7
Author: Bjorn Helgaas <bhelgaas@google.com>
Date:   Fri Mar 7 13:51:12 2014 -0700

    PCI: Change pci_bus_alloc_resource() type_mask to unsigned long

    The pci_bus_alloc_resource() "type_mask" parameter is used to compare with
    the "flags" member of a struct resource, so it should be the same type,
    namely "unsigned long".

    No functional change because all current IORESOURCE_* flags fit in 32 bits.

    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/bus.c b/drivers/pci/bus.c
index e3bdc88..fb8aed3 100644
--- a/drivers/pci/bus.c
+++ b/drivers/pci/bus.c
@@ -132,7 +132,7 @@ static void pci_clip_resource_to_region(struct pci_bus *bus,
 
 static int pci_bus_alloc_from_region(struct pci_bus *bus, struct resource *res,
   resource_size_t size, resource_size_t align,
-  resource_size_t min, unsigned int type_mask,
+  resource_size_t min, unsigned long type_mask,
   resource_size_t (*alignf)(void *,
        const struct resource *,
        resource_size_t,
@@ -200,7 +200,7 @@ static int pci_bus_alloc_from_region(struct pci_bus *bus, struct resource *res,
  */
 int pci_bus_alloc_resource(struct pci_bus *bus, struct resource *res,
   resource_size_t size, resource_size_t align,
-  resource_size_t min, unsigned int type_mask,
+  resource_size_t min, unsigned long type_mask,
   resource_size_t (*alignf)(void *,
        const struct resource *,
        resource_size_t,
diff --git a/include/linux/pci.h b/include/linux/pci.h
index 2c99ea1..0a3f2f5 100644
--- a/include/linux/pci.h
+++ b/include/linux/pci.h
@@ -1080,7 +1080,7 @@ void pci_bus_remove_resources(struct pci_bus *bus);
 int __must_check pci_bus_alloc_resource(struct pci_bus *bus,
    struct resource *res, resource_size_t size,
    resource_size_t align, resource_size_t min,
-   unsigned int type_mask,
+   unsigned long type_mask,
    resource_size_t (*alignf)(void *,
         const struct resource *,
         resource_size_t,
-- 
1.7.1